### PR TITLE
fix: fix the flex gap problem

### PR DESCRIPTION
**Pull request by nova1751**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [x] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

#48505

### 💡 Background and solution

Related to PR #48091.
Css attribute `gap` in `flex` item still takes effect when the DOM `element` size is set to 0. Setting a negative margin can solve this problem.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix the upload picture card animation issue     |
| 🇨🇳 Chinese |     修复`upload`组件图片卡的动画效果      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by nova1751** [comment]
Smooth again. 🥳

https://github.com/ant-design/ant-design/assets/97817985/189abfd5-18ea-4c21-b5c6-f2a07858e0a1

**Review by afc163 — APPROVED** [review]

File: components/upload/style/motion.ts [hunk 1]
```diff
@@ -4,30 +4,31 @@ import type { UploadToken } from '.';
...
   const { componentCls } = token;
-  const inlineCls = `${componentCls}-animate-inline`;
 
+  const uploadAnimateInlineIn = new Keyframes('uploadAnimateInlineIn', {
+    from: {
+      width: 0,
+      height: 0,
+      padding: 0,
+      opacity: 0,
+      margin: `calc(${token.marginXS} / -2)`,
```
Comment (MadCcc) [inline components/upload/style/motion.ts:17]: token.calc(token.marginXS).div(-2).equal()

Comment (MadCcc) [reply components/upload/style/motion.ts:17]: 数值模式下会直接计算

Comment (nova1751) [reply components/upload/style/motion.ts:17]: So should I make a new PR for this?

**Pull request merged**